### PR TITLE
WIP: Well things to std::map/vector

### PR DESCRIPTION
**Pull request by stefoss23**
**Task**
Sum well-stuff parameters converted to std.

**Pre un-WIP checklist**
- [ ] Statoil tests pass locally

### CODE REVIEW

**Comment by abrautomate** [comment]
Can one of the admins verify this patch?

**Comment by joakim-hove** [comment]
ok to test

File: lib/ecl/well_branch_collection.cpp [hunk 1]
```diff
@@ -81,16 +88,16 @@ bool well_branch_collection_has_branch( const well_branch_collection_type * bran
 
 
 const well_segment_type * well_branch_collection_iget_start_segment( const well_branch_collection_type * branches , int index ) {
-  if (index < vector_get_size( branches->__start_segments))
-    return (const well_segment_type*)vector_iget_const( branches->__start_segments , index);
+  if (index < (int)branches->__start_segments.size())
```
Comment (joakim-hove) [inline lib/ecl/well_branch_collection.cpp]: C++ way: `static_cast<int>(branches)`

File: lib/ecl/well_branch_collection.cpp [hunk 2]
```diff
@@ -33,8 +35,8 @@
 struct well_branch_collection_struct {
   UTIL_TYPE_ID_DECLARATION;
 
-  vector_type     *  __start_segments;
-  int_vector_type *  index_map;
+  std::vector<const well_segment_type*> __start_segments;
```
Comment (joakim-hove) [inline lib/ecl/well_branch_collection.cpp]: There is *something* with `const` members in STL containers; please remove the `const`.

File: lib/ecl/well_branch_collection.cpp [hunk 3]
```diff
@@ -18,6 +18,8 @@
 
 #include <stdbool.h>
 
+#include <vector>
+
 #include <ert/util/util.h>
 #include <ert/util/type_macros.hpp>
 #include <ert/util/vector.hpp>
```
Comment (joakim-hove) [inline lib/ecl/well_branch_collection.cpp]: I guess you can remove the `#include <ert/util/vector.hpp>`?

File: lib/ecl/well_conn_collection.cpp [hunk 4]
```diff
@@ -18,6 +18,8 @@
 
 #include <stdbool.h>
 
+#include <vector>
+
 #include <ert/util/util.h>
 #include <ert/util/type_macros.hpp>
 #include <ert/util/vector.hpp>
```
Comment (joakim-hove) [inline lib/ecl/well_conn_collection.cpp]: The `#include <ert/util/vector.hpp>` should be removed.

File: lib/ecl/well_info.cpp [hunk 5]
```diff
@@ -21,6 +21,8 @@
 
 #include <string>
 #include <vector>
+#include <map>
+#include <algorithm>
 
 #include <ert/util/util.h>
 #include <ert/util/hash.hpp>
```
Comment (joakim-hove) [inline lib/ecl/well_info.cpp]: The `#include <ert/util/hash.hpp>` can be removed.

**Comment by andreabrambilla** [comment]
The comment about verifying the patch from @abrautomate was unexpected. It is possible that this PR was created while Lars and I were tinkering with the jobs on Jenkins. Let us know if something like that pops up again

**Pull request merged**